include/circt/Dialect/FIRRTL/FIRRTLAttributes.td

```diff
@@ -229,4 +229,10 @@ def InternalPathAttr : AttrDef<FIRRTLDialect, "InternalPath"> {
 def InternalPathArrayAttr
   : TypedArrayAttrBase<InternalPathAttr, "InternalPath array attribute">;
 
+//===----------------------------------------------------------------------===//
+// Miscellaneous attributes
+//===----------------------------------------------------------------------===//
+
+def LayerArrayAttr : TypedArrayAttrBase<SymbolRefAttr, "an array of layers">;
```

Comment:
Might want to use ArrayRefAttr, which IMHO has a better getter: https://github.com/llvm/circt/blob/main/include/circt/Types.td#L32

Reply:
This would be better, but should add a `TypedArrayRefAttrBase`. I can include that.

Reply:
Landed that here: https://github.com/llvm/circt/commit/eac2a5e6114c362e848edae2fa687c6ef42541e6 Switching to use that in this PR.